samples/meeting-av-filters/js/ClientApp/src/pages/meeting-av-filter.jsx

```diff
@@ -0,0 +1,137 @@
+                    videoFrame.data[i + 1] = appliedEffect.pixelValue;
+                }
+            }
+
+            // Create a canvas for processing video frames.
+            let canvas = new OffscreenCanvas(480, 360);
+            let videoFilter = new WebglVideoFilter(canvas);
+            videoFilter.init();
+
+            // Process video frames for effect 2
+            function videoFrameHandler(videoFrame, notifyVideoProcessed, notifyError) {
+                switch (selectedEffectId) {
+                    case effectIds.half:
+                        simpleHalfEffect(videoFrame);
+                        break;
+                    case effectIds.gray:
+                        videoFilter.processVideoFrame(videoFrame);
+                        break;
+                    default:
+                        break;
+                }
+
+                //send notification the effect processing is finshed.
+                notifyVideoProcessed();
+            }
+
+            // Clear the selected effect
+            function clearSelect() {
+                document.getElementById("filter-half").classList.remove("selected");
+                document.getElementById("filter-gray").classList.remove("selected");
+            }
+
+            // This method is invoked when filter is changed
+            function effectParameterChanged(effectId) {
+                alert("effect changed" + effectId);
+                console.log(effectId);
+                if (selectedEffectId === effectId) {
+                    console.log('effect not changed');
+                    return;
+                }
+                
+                selectedEffectId = effectId;
+
+                // Clear the css for selected filter.
+                clearSelect();
+                switch (selectedEffectId) {
+                    case effectIds.half:
+                        console.log('current effect: half');
+                        document.getElementById("filter-half").classList.add("selected");
+                        break;
+                    case effectIds.gray:
+                        console.log('current effect: gray');
+                        document.getElementById("filter-gray").classList.add("selected");
+                        break;
+                    default:
+                        console.log('effect cleared');
+                        break;
+                }
+            }
+
+            // Register an handler to handle effect change event
+            video.registerForVideoEffect(effectParameterChanged);
+
+            // Register an handler to handle video frames
+            video.registerForVideoFrame(videoFrameHandler, {
+                format: 0,
+            });
+
+            // Any changes to the UI filter should be notified back to Teams client.
+            const filterHalf = document.getElementById("filter-half");
```

Comment:
If we're changing the sample to utilize `React`, we should get rid of using DOM apis directly.

Reply:
Updated code to use useRef hook